bayesian_cnn_prometheus/preprocessing/data_generator.py

```diff
@@ -30,12 +41,26 @@ def __init__(self, preprocessing_config: Dict, batch_size: int):
         self.image_loader = ImageLoader(
             preprocessing_config['transform_nifti_to_npy']['ext'])
 
-        self.config = DataGeneratorConfig(
+        self.config = config = DataGeneratorConfig(
             **preprocessing_config['create_chunks'])
 
         self.data_splitter = DataSplitter(preprocessing_config['create_data_structure'],
                                           preprocessing_config['update_healthy_patients_indices'])
-        self.dataset_structure = None
+        self.dataset_structure: Optional[Dict[str, List[str]]] = None
```

Comment:
Tu możemy dać coś takiego bo i tak w inicie nie da się przekazać dataset_structure więc to zawsze będzie generowane przez tego data spliettera, a wtedy możemy pozbyć się ifa w 75 linijce (który zawsze jest spełniony) i assert w 104.
```suggestion
        self.dataset_structure = self.data_splitter.split_indices()
```

Reply:
Masz rację